src/app/collective-rewards/allocations/context/utils.ts

```diff
@@ -27,3 +28,29 @@ export const validateAllocationsState = ({
 
   return true
 }
+
+export type GetVoteAllocationsParams = {
+  initialAllocations: Allocations
+  currentAllocations: Allocations
+  getBuilder: (address: Address) => Builder | null
+}
+export const getVoteAllocations = ({
+  initialAllocations,
+  currentAllocations,
+  getBuilder,
+}: GetVoteAllocationsParams) => {
+  return Object.entries(currentAllocations).reduce<[Address[], bigint[]]>(
+    (acc, [key, value]) => {
+      const builderAddress = key as Address
+      if (value !== initialAllocations[builderAddress]) {
+        const gauge = getBuilder(builderAddress)?.gauge
+        if (gauge) {
+          acc[0].push(gauge)
```

Comment:
not a biggie but prefer immutability whenever you can `acc[0] = [...acc[0], gauge]`